autocompleter/base.py

```diff
@@ -574,37 +574,47 @@ def suggest(self, term, facets=[]):
             for term_result_key in term_result_keys:
                 pipe.delete(term_result_key)
 
+            provider_facet_keys = provider.get_facets()
+            facet_key_mismatch = False
+
             if len(facets) > 0:
-                facet_result_keys = []
-                for facet in facets:
-                    try:
-                        facet_type = facet['type']
-                        if facet_type not in ['and', 'or']:
+                # we use from_iterable to flatten the list comprehension into a single list
```

Comment:
i don't love this approach because you are doing this per-provider when the suggest facet key set thing does not change per provider at all... and I don't love the extra level of nesting either.

instead of this approach I would do BEFORE the loop
```
facet_keys = set()
if len(facets):
    sub_facets = itertools.chain.from_iterable([facet['facets'] for facet in facets])
    facet_keys_set = set([sub_facet['key'] for sub_facet in sub_facets])
  ```

then inside the loops
```
use_facets = False
if len(facet_keys):
    provider_keys_set = set(provider.get_facets())
     if facet_keys_set.issubset(provider_keys_set):
         use_facets = True

if use_facets:
    ...
```